packages/web/src/components/createCell.tsx

```diff
@@ -77,6 +77,12 @@ type Guaranteed<T> = {
   [K in keyof T]-?: NonNullable<T[K]>
 }
 
+type KeyCount<T extends object> = L.Length<U.ListOf<O.SelectKeys<T, any>>>
```

Comment:
I wasn't sure whether there was a reliable way to count keys/properties like this, but I managed to cobble this together.  I haven't found any simpler way to do this that made the new tests pass.